src/ripple/test/jtx/Envss.h

```diff
@@ -0,0 +1,96 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_TEST_JTX_ENVSS_H_INCLUDED
+#define RIPPLE_TEST_JTX_ENVSS_H_INCLUDED
+
+#include <ripple/test/jtx/Env.h>
+
+namespace ripple {
+namespace test {
+
+//------------------------------------------------------------------------------
+
+namespace jtx {
+
+//------------------------------------------------------------------------------
+
+/** A transaction testing environment.
+    Transactions submitted in sign-and-submit mode
+    by default.
+*/
+class Envss
+{
+private:
+    Env& env_;
+
+public:
+
+    class SignSubmitRunner
+    {
+    public:
+        SignSubmitRunner() = delete;
+        SignSubmitRunner(SignSubmitRunner const&) = delete;
+        SignSubmitRunner& operator= (SignSubmitRunner const&) = delete;
+        SignSubmitRunner(SignSubmitRunner&&) = default;
+        SignSubmitRunner& operator= (SignSubmitRunner&&) = default;
```

Comment:
Fwiw, all of these could be omitted except for the move constructor.  I don't feel strongly that they should, this is just fyi.
The move assignment operator would be a little easier to read if it was `= delete;`.  `SignSubmitRunner` isn't move assignable because of the reference data member on line 66 and the const data member on line 67.